packages/java-shell/src/main/kotlin/com/mongodb/mongosh/service/Cursor.kt

```diff
@@ -10,7 +10,9 @@ import org.graalvm.polyglot.Value
 
 internal class Cursor(private var helper: BaseMongoIterableHelper<*>, private val converter: MongoShellConverter, private val wrapper: ValueWrapper) : ServiceProviderCursor {
     private var iterator: MongoCursor<out Any?>? = null
-    private var closed = false
+
+    @HostAccess.Export
+    override var closed = false
```

Comment:
By default Kotlin will generate a private field with getter and setter, so you need `@JvmField` annotation to force it to be a public field.
But this annotation is not applicable to overridden properties, so you'll need to remove the property from `ServiceProviderCursor` inteface.
```suggestion
    @JvmField
    @HostAccess.Export
    override var closed = false
```